src/nvim/api/extmark.h

```diff
@@ -4,14 +4,27 @@
 
 #include "nvim/api/keysets_defs.h"  // IWYU pragma: keep
 #include "nvim/api/private/defs.h"  // IWYU pragma: keep
+#include "nvim/buffer_defs.h"
 #include "nvim/decoration_defs.h"  // IWYU pragma: keep
 #include "nvim/macros_defs.h"
 #include "nvim/map_defs.h"
 #include "nvim/types_defs.h"
 
 EXTERN Map(String, int) namespace_ids INIT( = MAP_INIT);
+/// Non-global namespaces.
```

Comment:
Let's explicitly document this quirk, until we decide what to do about it:

```suggestion
/// Non-global namespaces. A locally-scoped namespace may be "orphaned" if all
/// window(s) it was scoped to, are destroyed.
```